lxc/copy.go

```diff
@@ -153,6 +154,9 @@ func (c *cmdCopy) copyInstance(conf *config.Config, sourceResource string, destR
 			return fmt.Errorf(i18n.G("--instance-only can't be passed when the source is a snapshot"))
 		}
 
+		// Remove snapshot suffix from destination name.
```

Comment:
This comment say *what* but doesn't explain the *why*.

Reply:
Thanks for this one.

Snapshot suffix is removed to ensure consistency when copying/moving an instance (or snapshot) to a different remote (`lxc move c1 remote2:` or `lxc move c1/snap0 remote2:`).

However, I missed the fact, that this will trim the suffix even if the user provides the snapshot name in the destination name.

For example, the following commands produced the same error, although they should not: 
```
lxc copy c1/snap0 c1/snap1
lxc copy c1/snap0 c1

# Error: Failed creating instance record: Add instance info to the database: This "instances" entry already exists`
```

Now, the snapshot name is trimmed only when destName is set from the source.

---

A bit unrelated, but is there a reason we do not output the instance name in the above error? (something like `Database entry for instance "c1" already exists`)